packages/wonder-blocks-dropdown/src/components/single-select.tsx

```diff
@@ -267,8 +321,44 @@ const SingleSelect = (props: Props) => {
         if (onToggle) {
             onToggle(opened);
         }
+        if (!opened && required && !selectedValue) {
+            // If closed, field is required, and no value is selected, validate
+            handleValidation(selectedValue);
+        }
     };
 
+    const handleValidation = React.useCallback(
```

Comment:
Will work on refactoring shared logic between SingleSelect and MultiSelect in the next PR!